### PR TITLE
Fix hall of fame images

### PR DESCRIPTION
**Pull request by Ultimatum22**
I kind of rewrote the logic to be more generic and account for changes on formula1 side, as long as they don't change the html.

I was a bit confused about the endpoint check because the one from Hive is never the same as the one set hardcoded in formula_one.dart, because the default values differ. I left it the same for now.

### CODE REVIEW

**Comment by BrightDV** [comment]
Thanks for the PR! The code is a lot more cleaner that what I have done... (I'm not proud of it though)
Also, can you add a screenshot of the working list?
> I was a bit confused about the endpoint check because the one from Hive is never the same as the one set hardcoded in formula_one.dart, because the default values differ. I left it the same for now.

I know that it is a bit confusing, but I created the server to support the web version (mostly). So, when using a boxbox server, the API url has to be changed, but also the main website url, as it is blocked by the CORS too. So when the server is changed (e.g., the endpoint here), the main website url is changed too.
I hope that it is clear enough.

File: lib/scraping/formula_one.dart [hunk 1]
```diff
@@ -564,38 +565,31 @@ class FormulaOneScraper {
...
+      dom.Document document = parser.parse(response.body);
+      List<dom.Element> elements = document.querySelectorAll('a.column.column-4[href*="/en/drivers/hall-of-fame/"][href\$=".html"]');
+
+      for (dom.Element element in elements) {
+
+        List<String> driverInfo = element.getElementsByClassName('teaser-info-title').first.text.trim().split(' - ');
+        String driverName = driverInfo[0];
+        String driverYears = driverInfo[1];
+
+        dom.Element imageElement = element.getElementsByTagName('img').firstWhere((e) => e.classes.contains('hidden'));
```
Comment (BrightDV) [inline lib/scraping/formula_one.dart:584]: Can you add [a trailing comma](https://docs.flutter.dev/tools/formatting#using-trailing-commas) and then run a `dart format . --set-exit-if-changed` to format the code?

**Review by BrightDV — CHANGES_REQUESTED** [review]

File: lib/scraping/formula_one.dart (hunk 1 above)
Comment (Ultimatum22) [reply lib/scraping/formula_one.dart]: I don't get the trailing icon here. I did applied the formatter, I was thinking of adding it to a workflow to fix this on the fly.

**Comment by Ultimatum22** [comment]
Ah damn, sorry forgot about the screenshot.

**Comment by Ultimatum22** [comment]
> Thanks for the PR! The code is a lot more cleaner that what I have done... (I'm not proud of it though) Also, can you add a screenshot of the working list?
> 
> > I was a bit confused about the endpoint check because the one from Hive is never the same as the one set hardcoded in formula_one.dart, because the default values differ. I left it the same for now.
> 
> I know that it is a bit confusing, but I created the server to support the web version (mostly). So, when using a boxbox server, the API url has to be changed, but also the main website url, as it is blocked by the CORS too. So when the server is changed (e.g., the endpoint here), the main website url is changed too. I hope that it is clear enough.

Ah that does make sense, only the endpoint variable set now is never used so always different.

I can refactor the rest of the class if you want me too.

**Comment by BrightDV** [comment]
> Ah that does make sense, only the endpoint variable set now is never used so always different.
> 
> I can refactor the rest of the class if you want me too.

You can if you want to, but I think that a rewrite would be more appropriate when a function breaks, like this one. I'm not sure if it is a really good way to do, but at least it is easier.
Anyway, thanks for your the PR & your efforts.

File: lib/scraping/formula_one.dart (hunk 1 above)
Comment (BrightDV) [reply lib/scraping/formula_one.dart]: > I don't get the trailing icon here. I did applied the formatter, I was thinking of adding it to a workflow to fix this on the fly.

Yes, no worries. I will do a proper workflow for the PRs with formatting and analysing .

**Pull request merged**